design-proposals/testing-and-ci/properly-using-context-in-functional-tests.md

```diff
@@ -0,0 +1,133 @@
+# Properly use contexts in functional tests
+
+## Overview
+This proposal is about properly use golang contexts in KubeVirt functional tests.
+
+## Introduction
+Since the start of the KubeVirt project, contexts became a very important concept
+in golang, when writing an asynchronous logic. In a way, the KubeVirt code left
+behind regarding the proper usage of golang contexts.
+
+Using `context.Background()` or even worse, `context.TODO()` is a code smell, because
+doing that won't allow the application to cancel async-operation, like I/O, for example,
+in case of termination.
+
+Closing this gap is a huge effort. Contexts should be passed deeper and deeper down
+the call stack, changing the api of many functions and methods. The impact is meaningful.
```

Comment:
Are we talking about the actual code or the test code here?

Reply:
In which way is the impact meaningful? Measurable time difference? Less hangs?

Reply:
What I was trying to say is that in production code, it will be very hard to move contexts around, and it much easier to do that in the functional tests, so let's start there

Reply:
re-wrote the whole paragraph.